src/PlacesAutocomplete.js

```diff
@@ -47,17 +47,20 @@ class PlacesAutocomplete extends Component {
       secondaryText: structured_formatting.secondary_text,
     })
 
-    const { highlightFirstSuggestion } = this.props
+    const { highlightFirstSuggestion, filterPredictions } = this.props
 
-    this.setState({
-      autocompleteItems: predictions.map((p, idx) => ({
-        suggestion: p.description,
-        placeId: p.place_id,
-        active: (highlightFirstSuggestion && idx === 0 ? true : false),
-        index: idx,
-        formattedSuggestion: formattedSuggestion(p.structured_formatting),
-      }))
-    })
+    const filteredPredictions = filterPredictions ?
+      predictions.filter(filterPredictions) : predictions
+
+    const autocompleteItems = filteredPredictions.map((p, idx) => ({
```

Comment:
With the suggested default `filteredPrediction` we can simply chain on functions here, and remove temporary local variable `const filteredPredictions`.

```js
predictions.filter(filterPredictions).map((p, idx) => {
  // formatting suggestion
});
```